Civi/Selfservice/ActionProvider/Action/GetHash.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/*------------------------------------------------------------+
+| Selfservice extension                                       |
+| Copyright (C) 2019 SYSTOPIA                                 |
+| Author: B. Endres (endres@systopia.de)                      |
++-------------------------------------------------------------+
+| This program is released as free software under the         |
+| Affero GPL license. You can redistribute it and/or          |
+| modify it under the terms of this license which you         |
+| can read by viewing the included agpl.txt or online         |
+| at www.gnu.org/licenses/agpl.html. Removal of this          |
+| copyright header is strictly prohibited without             |
+| written permission from the original author(s).             |
++-------------------------------------------------------------*/
+
+namespace Civi\Selfservice\ActionProvider\Action;
+
+use CRM_Selfservice_ExtensionUtil as E;
+
+use \Civi\ActionProvider\Action\AbstractAction;
+use \Civi\ActionProvider\Parameter\ParameterBagInterface;
+use \Civi\ActionProvider\Parameter\Specification;
+use \Civi\ActionProvider\Parameter\SpecificationBag;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class GetHash extends AbstractAction implements CompilerPassInterface {
```

Comment:
Maybe we can use ``token`` or ``checksum`` in the UI instead of ``hash``, so people know it's *not* the (secret) contact hash...

Reply:
I changed the name to token